api/api/src/main/scala/uk/ac/wellcome/platform/api/swagger/SwaggerScalaModelConverter.scala

```diff
@@ -56,6 +70,19 @@ class SwaggerScalaModelConverter extends AbstractModelConverter(Json.mapper()) {
     annotatedType.getType match {
       case rt: ReferenceType      => rt.getContentType
       case ct: CollectionLikeType => ct.getContentType
+      case st: SimpleType         => st.containedType(0)
       case t                      => throw new IllegalArgumentException(s"Unknown type $t")
     }
+
+  private def getAggregationClassName(annotatedType: AnnotatedType): String = {
+    val cls = getClass(getContentType(annotatedType))
+    if (cls == classOf[DisplayGenre]) "Genre"
+    else if (cls == classOf[DisplayWorkType]) "WorkType"
+    else if (cls == classOf[DisplayPeriod]) "Period"
+    else if (cls == classOf[DisplaySubject]) "Subject"
+    else if (cls == classOf[DisplayLanguage]) "Language"
+    else throw new IllegalArgumentException(s"Unknown class $cls")
```

Comment:
I know generally explicit is better - but would a check on `statsWith("Display")` and just replacing that work?

Reply:
Not just explicitness here, imo it much more robust working with the types themselves rather than strings (using strings will include other models that we don't want to include etc)